include/envoy/server/admin.h

```diff
@@ -23,8 +24,8 @@ namespace Server {
  */
 #define MAKE_ADMIN_HANDLER(X)                                                                      \
   [this](absl::string_view path_and_query, Http::HeaderMap& response_headers,                      \
-         Buffer::Instance& data) -> Http::Code {                                                   \
-    return X(path_and_query, response_headers, data);                                              \
+         Buffer::Instance& data, Http::StreamDecoderFilterCallbacks* callbacks) -> Http::Code {    \
```

Comment:
As mentioned earlier, IMO all these args could be saved in the AdminFilter object -- and the callbacks already are.  That would simplify this interface once and for all, rather than adding a very specialized complicated object to all of them.